packages/vm/src/evm/types.ts

```diff
@@ -292,3 +293,10 @@ type EVMResult = {
 export interface EVMInterface {
   runMessage(environment: EVMEnvironment): Promise<EVMResult>
 }
+
+export type EVMEvents = {
+  newContract: (data: NewContractEvent, resolve?: (result: any) => void) => void
+  beforeMessage: (data: Message, resolve?: (result: any) => void) => void
+  afterTx: (data: EVMResult, resolve?: (result: any) => void) => void
+  step: (data: InterpreterStep, resolve?: (result: any) => void) => void
+}
```

Comment:
Ok, besides all the fear I have that too much stuff is getting into this PR I have to say:

This is a *dramatic* improvement on the current status quo!!! 🎉 🙏 💯

Reply:
Good catch, I copied these function sigs around and forgot to remove `afterTx`

Reply:
And yes, the types are absolutely fantastic :heart_eyes: